source/slang/slang.cpp

```diff
@@ -561,17 +561,15 @@ void CompileRequest::generateIR()
     // in isolation.
     for( auto& translationUnit : translationUnits )
     {
-        // TODO JS:
-        // This is a bit of HACK. Apparently if we call generateIRForTranslationUnit(translationUnit) twice
-        // we get a different result (!).
-        // So here, we only create once even if we run verification.
-        RefPtr<IRModule> irModule;
+        // We want to only run generateIRForTranslationUnit once here. This is for two side effects:
+        // * it can dump ir 
+        // * it can generate diagnostics
+
+        /// Generate IR for translation unit
+        RefPtr<IRModule> irModule(generateIRForTranslationUnit(translationUnit));
```

Comment:
Incidental: as a style thing I personally dislike the C++ constructor initialization syntax (`Type name(initVal);`) because it is difficult to parse visually comapred to more conventional C notation (`Type name = initVal;`).

For my own education, is this a case where using the latter syntax actually introduces a performance overhead (extra reference-counting work)? Is that the kind of overhead that the stricter rules around copy elision in more recent C++ versions will eliminate?

Reply:
I also think doing = is much more readable. 
Unfortunately compilers can do do more work if you do that style tho - at least when Iooked into just that a few years ago they definitely did. Right now though things may have improved, I'd have to check.